stdlib/ostruct.rb

```diff
@@ -15,26 +17,33 @@ def []=(name, value)
     @table[new_ostruct_member(name)] = value
   end
 
-  def method_missing(name, *args)
+  def method_missing(mid, *args)
     if args.length > 2
-      raise NoMethodError.new("undefined method `#{name}' for #<OpenStruct>", name)
+      raise NoMethodError.new("undefined method `#{mid}' for #<OpenStruct>", mid)
     end
-    if name.end_with? '='
+    if mid.end_with? '='
       if args.length != 1
-        raise ArgumentError.new "wrong number of arguments (0 for 1)"
+        raise ArgumentError, 'wrong number of arguments (0 for 1)'
       end
-      @table[new_ostruct_member(name[0 .. -2])] = args[0]
+      @table[new_ostruct_member(mid[0..-2])] = args[0]
+    elsif args.empty?
+      @table[mid.to_sym]
     else
-      @table[name.to_sym]
+      super
```

Comment:
Is this an unrelated fix? If so maybe it's worth its own PR or going straight to master

Reply:
That's what MRI does. Would you like me to revert it, add to filters for now and send it in a separate PR?

Reply:
@iliabylich commit straight to master, the diff in this PR should become clear after that